service/history/workflow/update/export_test.go

```diff
@@ -32,3 +32,10 @@ var (
 	NewAccepted  = newAccepted
 	NewCompleted = newCompleted
 )
+
+// ObserveCompletion exporses withOnComplete to unit tests
+//
+//revive:disable-next-line:unexported-return for testing
```

Comment:
I'm gonna side with revive here. They should both be exported, or neither of them should. Since we're using test packages, we should make the function and return type exported. That doesn't mean we are allowing users to make their own `UpdateOpt` options